src/psyclone/dynamo0p3.py

```diff
@@ -8261,18 +8306,21 @@ def ncolours_var(self):
                                colour-map information has not been initialised.
         '''
         if not self.is_coloured():
-            raise InternalError("Kernel '{0}' is not inside a coloured "
-                                "loop.".format(self.name))
+            raise InternalError(f"Kernel '{self.name}' is not inside a "
+                                f"coloured loop.")
         if self._is_intergrid:
             invoke = self.ancestor(InvokeSchedule).invoke
             if id(self) not in invoke.meshes.intergrid_kernels:
                 raise InternalError(
-                    "Colourmap information for kernel '{0}' has not yet "
-                    "been initialised".format(self.name))
-            ncols = invoke.meshes.intergrid_kernels[id(self)].ncolours_var
-        else:
-            ncols = self.scope.symbol_table.lookup_with_tag("ncolour").name
-        return ncols
+                    f"Colourmap information for kernel '{self.name}' has "
+                    f"not yet been initialised")
+            ncols_sym = \
+                invoke.meshes.intergrid_kernels[id(self)].ncolours_var_symbol
+            if not ncols_sym:
+                return None
```

Comment:
Since this routine can now return None the docstring needs updating.

Reply:
Done.